File: Arise.FileSyncer.Service/Ipc/Messages/ReceivedProfileMessage.cs
using System;
using Arise.FileSyncer.Core;

namespace Arise.FileSyncer.Service.Ipc.Messages
{
    public class ReceivedProfileMessage : IpcMessage
    {
        public override string Command => "receivedProfile";

        public Guid ConnectionId { get; set; }
        public string Id { get; set; }
        public Guid Key { get; set; }
        public string Name { get; set; }
        public DateTime CreationDate { get; set; }
        public bool SkipHidden { get; set; }

        internal IpcMessage Fill(ProfileReceivedEventArgs e)
        {
            ConnectionId = e.RemoteId;
            Id = e.ProfileShare.Id.ToString();
            Key = e.ProfileShare.Key;
            Name = e.ProfileShare.Name;
            CreationDate = e.ProfileShare.CreationDate;
            SkipHidden = e.ProfileShare.SkipHidden;
            return this;
        }
    }
}

File: Arise.FileSyncer.TauriUI/src/ipc/messages/receivedProfileMessage.ts
import { IpcMessage } from "../ipcMessage"

export const Command = "receivedProfile"

export interface Message extends IpcMessage, Data { }

export interface Data {
    ConnectionId: string,
    Id: string,
    Key: string,
    Name: string,
    CreationDate: string,
    SkipHidden: boolean,
}
